Server.Core/Request.cs

```diff
@@ -6,15 +6,15 @@ public class Request
     {
         public string HttpVersion { get; internal set; }
         public string Uri { get; internal set; }
-        public string Method { get; internal set; }
+        public HttpMethod Method { get; internal set; }
```

Comment:
They are internal because I want them to be able to be used by the builder, but not anything outside of the framework.  I don't understand why you would use private setters since nothing could access them.